core/src/com/unciv/ui/worldscreen/Minimap.kt

```diff
@@ -120,13 +123,65 @@ class Minimap(val mapHolder: WorldMapHolder) : Table(){
 class MinimapHolder(mapHolder: WorldMapHolder): Table() {
     val minimap = Minimap(mapHolder)
     val worldScreen = mapHolder.worldScreen
+    private data class ToggleButtonInfo(val actor: Actor, val getIt: ()->Boolean, val setIt: (Boolean)->Unit)
+    private val toggleButtonInfo: EnumMap<MinimapToggleButtons,ToggleButtonInfo> = EnumMap<MinimapToggleButtons,ToggleButtonInfo>(MinimapToggleButtons::class.java)
 
     init {
         add(getToggleIcons()).align(Align.bottom)
         add(getWrappedMinimap())
         pack()
     }
 
+    enum class MinimapToggleButtons {
```

Comment:
enum class MinimapToggleButtons (val icon: String) {
    YIELD ("Food"),
    WORKED ("Population"),
...
}

Reply:
Yes, shorter - but I thought this might incur more runtime overhead...